### PR TITLE
Store dt in parameter set for microphysics parameter

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR fixes a performance bug. See [this comment](https://github.com/CliMA/ClimaAtmos.jl/pull/353#discussion_r860327896). cc @szy21

### CODE REVIEW

**Review by szy21 — APPROVED** [review]

**Comment by szy21** [comment]
hmm why should this change the mse?

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl [hunk 1]
```diff
@@ -6,15 +6,18 @@ const CM = CloudMicrophysics
 
 include("../staggered_nonhydrostatic_model.jl")
 
-struct BaroclinicWaveParameterSet <: AbstractEarthParameterSet end
+struct BaroclinicWaveParameterSet{NT} <: AbstractEarthParameterSet
+    nt::NT
```
Comment (jiahe23) [inline examples/hybrid/sphere/baroclinic_wave_utilities.jl:10]: while you fix the buildkite failure, can we also think of a better property name here? I do not understand what `nt` referes to

Comment (charleskawczynski) [reply examples/hybrid/sphere/baroclinic_wave_utilities.jl:10]: Agreed, how about `named_tuple`?

**Comment by charleskawczynski** [comment]
> hmm why should this change the mse?

@jiahe23 pointed out that we need to rebase (the mse tables have changed, so we're no longer inside the tolerance). I think all we need to do is rebase, I'll give that a shot.

**Comment by szy21** [comment]
Ah, maybe I shouldn't have changed the mse table last time. The mse is within the tolerance so the ci passed, but I updated the table anyway: https://github.com/CliMA/ClimaAtmos.jl/pull/356/files. Should I change it back?

**Comment by charleskawczynski** [comment]
> Ah, maybe I shouldn't have changed the mse table last time. The mse is within the tolerance so the ci passed, but I updated the table anyway: https://github.com/CliMA/ClimaAtmos.jl/pull/356/files. Should I change it back?

It's totally fine, the whole point is just to help identify / indicate when we do have behavioral changes--and in which cases.

**Comment by charleskawczynski** [comment]
> How is the tolerance determined though? Why is it zero for some examples but quite large for some other examples? Maybe I misunderstood the mse. Is it compared with the current main?

The tolerance is an [averaged normalized error per variable](https://github.com/CliMA/NCRegressionTests.jl/blob/main/src/NCRegressionTests.jl#L82-L91). We currently compare with the latest main branch, but we should move to a semi-fixed reference (#326) to avoid "floating" results. It won't be too difficult to make this change, I need to tweak a few pieces, and what we have in the mean time is still effective.

**Comment by szy21** [comment]
> We currently compare with the latest main branch

I see, I am not expecting any behavior change for this PR, so I should expect the mse to be all 0 after the rebase?

**Comment by charleskawczynski** [comment]
> I see, I am not expecting any behavior change for this PR, so I should expect the mse to be all 0 after the rebase?

Non-behavior changing PRs should not require _updating_ the mse table (they can be, and remain non-zero).

**Comment by szy21** [comment]
> > I see, I am not expecting any behavior change for this PR, so I should expect the mse to be all 0 after the rebase?
> 
> Non-behavior changing PRs should not require _updating_ the mse table (they can be, and remain non-zero).

Ok, thanks! Let's see what you get after the rebase.

**Comment by charleskawczynski** [comment]
Interesting, I need to take a closer look

**Comment by charleskawczynski** [comment]
I've updating the regression tests, and it should be simpler to determine if this is actually resulting in behavioral changes, since our reference dataset is now fixed (we can compare the reference here with the one in the latest merged branch).

**Comment by charleskawczynski** [comment]
Ok, it seems that this PR is changing the following cases:

 - held suarez (ρe) equilmoist
 - held suarez (ρe_int) equilmoist

I'm going to try a separate branch that simply sets `const dt = `, to see if it's somehow changing. It could also be a type conversion, I suppose?

**Comment by charleskawczynski** [comment]
Wow, this is definitely behavior changing. I looked at the results and compared against another branch where `dt` is set to a constant, but the changes are basically miniscule. It's possible that this somehow changes order of operations if the compiler can reason that the parameter is constant. I'm going to update the MSE tables and merge.

I think it would be helpful if we plotted figures that made comparisons a bit easier, comparing movies is difficult.

**Comment by charleskawczynski** [comment]
Hmm, I want to check one more thing... We should be able to merge this either way, but understanding what's going on will be helpful down the road.

**Comment by charleskawczynski** [comment]
It does seem like there are changes, it could be a reproducibility issue. See [this comment](https://github.com/CliMA/ClimaAtmos.jl/pull/307#issuecomment-1113934104).

The results still look fine, I'm updating the tables and we can merge.

**Pull request merged**